PublicLogs update to provide more information

ad-astra-video opened the pull request:
**What does this pull request do? Explain your changes. (required)**
Expands the PublicLogs information to provide insight into transcoding including when streams swap orchestrators, per segment performance and orchestrator selection.

**Specific updates (required)**

- **clog.go** - update keys to include "ethaddress" and "orchestrator" (url).  Tests updated for this as well.
- **broadcast.go** - Various log lines added.  Updates included to return data to central location to log information (suspending orchestrator, selected sessions).  Log lines enhanced when there are failures to distinguish what the failures were.
- **rpc.go** - BroadcastSession Address() function updated to return the on chain orchestrator address.  Currently returns the local eth address for an Orchestrator which for many Os is different than the onchain address. Tests updated.

- **segment_rpc.go** - SubmitSegment return errors enhanced to provide more insight to failures. Tests updated.

- **selection.go** - Updated to return information on why the orchestrator was selected (performance, random, stake). Tests updated.

**How did you test each of these updates (required)**
Running updates on test broadcaster


**Does this pull request close any open issues?**
No


**Checklist:**
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] Read the [contribution guide](./doc/contributing.md)
- [X] `make` runs successfully
- [X ] All tests in `./test.sh` pass
- [ ] README and other documentation updated
- [ ] [Pending changelog](./CHANGELOG_PENDING.md) updated

ad-astra-video committed:
update tests

ad-astra-video committed:
broadcaster introspection: log lines for selection and change of O

ad-astra-video committed:
add eth addr and orch to session completed log line

ad-astra-video committed:
remove 'broadcaster introspection' comments

ad-astra-video committed:
fix tests and revert Address() in rpc.go

thomshutt commented on server/selection.go:
```diff
@@ -16,7 +17,7 @@ const SELECTOR_LATENCY_SCORE_THRESHOLD = 1.0
 	Add(sessions []*BroadcastSession)
 	Complete(sess *BroadcastSession)
-	Select(ctx context.Context) *BroadcastSession
+	Select(ctx context.Context) (*BroadcastSession, string)
```
Could you add a comment about what this string is please

ad-astra-video committed:
update tests

ad-astra-video committed:
update verify.go sigVerification to return early if sess.Address() is nil instead of TicketParams

ad-astra-video committed:
update verify_test.go for change in early exit of sigVerification

ad-astra-video commented:
@thomshutt  I got the test.sh tests all working now I think. My local running of test.sh did not match up to the test github workflow.  My local runs would return a null error on a redeemer test and skip the rest of the tests it looks like. I do not see this same behavior in the github test workflow.

After fixing the updated error message tests, the tests linked below were failing because of how the local verification worked with signature failure errors.  I believe the tests below worked before because there was no `TicketParams` in the `OrchestratorInfo` in the `BroadcastSession` generated for testing. The `sigVerification` function exits early with no error if the `OrchestratorInfo` has no `TicketParams` in the `BroadcastSession` created for tests. https://github.com/livepeer/go-livepeer/blob/3f624087983b1ca0e161e7be0159d60b158b7727/verification/verify.go#L186-L189

With no changes to `verify.go` and `verify_test.go`, if I remove the `TicketParams` from the test `BroadcastSession` and the `sess.RecipientAddress()` from the `transcodeSeg` function the tests linked below pass.   

With my updates to add `ethaddress` to the PublicLogs I added a function in `BroadcastSession` to return the `Recipient` eth address for the logs.  I updated the `broadcast_test.go` tests to provide the `TicketParams` and `Recipient` where applicable to allow the `BroadcastSession` used in `transcodeSeg` function tests to have the data needed to complete the tests.  I did not think this would be a problem because my understanding is the pool building process kicks out sessions with no `TicketParams` when building the pool.
https://github.com/livepeer/go-livepeer/blob/3f624087983b1ca0e161e7be0159d60b158b7727/server/broadcast.go#L779-L783.  


**Tests having issues with after adding TicketParams with Recipient Address to BroadcastSession created for testing**
To get these tests to work I had to do downstream updates in `verify.go` and `verify_test.go`.  The fixes in  `verify.go` and `verify_test.go` were minor I believe.  I updated the `sigVerifcation` function to return early with no error if the `OrchestratorInfo.Address` is nil rather than the `TicketParams` and updated tests to in `verify_test.go` to match this change.

https://github.com/livepeer/go-livepeer/blob/3f624087983b1ca0e161e7be0159d60b158b7727/server/broadcast_test.go#L1226
https://github.com/livepeer/go-livepeer/blob/3f624087983b1ca0e161e7be0159d60b158b7727/server/broadcast_test.go#L1379
https://github.com/livepeer/go-livepeer/blob/3f624087983b1ca0e161e7be0159d60b158b7727/server/broadcast_test.go#L1537

ad-astra-video commented:
Closing this one to split into smaller PRs as requested.
